src/m365/spo/commands/group/group-member-remove.ts

```diff
@@ -74,25 +89,69 @@ class SpoGroupMemberRemoveCommand extends SpoCommand {
           return `Specified "groupId" ${args.options.groupId} is not valid`;
         }
 
+        if (args.options.userId && isNaN(args.options.userId)) {
+          return `Specified "userId" ${args.options.userId} is not valid`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid userName`;
+        }
+
+        if (args.options.email && !validation.isValidUserPrincipalName(args.options.email)) {
+          return `${args.options.email} is not a valid email`;
+        }
+
         return validation.isValidSharePointUrl(args.options.webUrl);
       }
     );
   }
 
   #initOptionSets(): void {
-    this.optionSets.push({ options: ['groupName', 'groupId'] });
+    this.optionSets.push(
+      { options: ['groupName', 'groupId'] },
+      { options: ['userName', 'email', 'userId'] }
+    );
+  }
+
+  private async getUserName(logger: Logger, args: CommandArgs): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving information about the user ${args.options.userName || args.options.email}`);
+    }
+
+    if (args.options.userName) {
+      return args.options.userName;
+    }
+
```

Comment:
Hi @nanddeepn, one last comment. De verbose logging statement should be here, after the 'short track' of returning the username. It's only requesting info if the username option has not been used, so the verbose line only makes sense after the short tracked return.